src/main/java/it/pagopa/pn/deliverypush/action/StartWorkflowHandler.java

```diff
@@ -59,38 +62,71 @@ public StartWorkflowHandler(
      */
     public void startWorkflow(String iun) {
         log.info("Start notification process iun={}", iun);
+        PnAuditLogBuilder auditLogBuilder = new PnAuditLogBuilder();
+        PnAuditLogEvent logEvent = auditLogBuilder
+                .before(PnAuditLogEventType.AUD_NT_INSERT, "start notification workflow for iun={}", iun)
+                .iun(iun)
+                .build();
+        logEvent.log();
         try {
             NotificationInt notification = notificationService.getNotificationByIun(iun);
 
             try {
                 //Validazione degli allegati della notifica
                 checkAttachmentUtils.validateAttachment(notification);
-
-                String legalFactId = saveLegalFactsService.saveNotificationReceivedLegalFact(notification);
-
-                addTimelineElement(timelineUtils.buildAcceptedRequestTimelineElement(notification, legalFactId), notification);
+                try {
+                    String legalFactId = saveLegalFactsService.saveNotificationReceivedLegalFact(notification);
+                    addTimelineElement(timelineUtils.buildAcceptedRequestTimelineElement(notification, legalFactId), notification);
+                    logEvent.generateSuccess().log();
+                } catch (Exception exc) {
+                    log.error("exception starting workflow", exc);
+                    logEvent.generateFailure("Exception on saveNotification", exc.getMessage()).log();
+                    throw exc;
+                }
 
                 //Start del workflow per ogni recipient della notifica
                 for (NotificationRecipientInt recipient : notification.getRecipients()) {
                     Integer recIndex = notificationUtils.getRecipientIndex(notification, recipient.getTaxId());
-                    startNotificationWorkflowForRecipient(notification, recIndex);
+                    PnAuditLogEvent logEventValid = auditLogBuilder
+                            .before(PnAuditLogEventType.AUD_NT_VALID, "Starting notification workflowforRecipient for recipient={}", recIndex)
+                            .iun(iun)
+                            .build();
+                    logEventValid.log();
+                    try {
+                        startNotificationWorkflowForRecipient(notification, recIndex);
+                        logEvent.generateSuccess("Starting notification workflowforRecipient for recipient={}", recIndex).log();   
+                    } catch (Exception exc) {
+                        logEventValid.generateFailure("Exception in Start notification workflow", exc).log();
+                        throw exc;
+                    }
                 }
             } catch (PnValidationException ex) {
                 handleValidationError(notification, ex);
             }
-        }catch (PnInternalException ex){
-            log.error("exception starting workflow", ex);
+        } catch (PnInternalException ex){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)